src/web/modules/agreements/agreements.http.ts

```diff
@@ -0,0 +1,66 @@
+import {NextFunction, Request, Response} from 'express'
+
+import {AdminUsers, Connector, Ledger} from '../../../lib/pay-request/client'
+import {EntityNotFoundError} from '../../../lib/errors'
+import logger from '../../../lib/logger'
+import {TransactionType} from "../../../lib/pay-request/shared";
+import {AgreementListFilterStatus, resolveAgreementStates} from "./states";
+
+const process = require('process')
+const url = require('url')
+const https = require('https')
+const moment = require('moment')
+const rfc822Validator = require('rfc822-validate')
+
+const {common, services} = require('./../../../config')
+
+if (common.development) {
+  process.env.NODE_TLS_REJECT_UNAUTHORIZED = 0
+}
+
+export async function list(req: Request, res: Response, next: NextFunction): Promise<void> {
+  try {
+    let service, account
+    const accountId = req.query.account
+    const selectedStatus = req.query.status as AgreementListFilterStatus || AgreementListFilterStatus.All
+    
+    const filters = {
+      ...req.query.reference && {reference: req.query.reference as string},
+      ...req.query.gateway_agreement_id && {gateway_agreement_id: req.query.gateway_agreement_id as string},
```

Comment:
Filter can likely be removed re: comment on the template

Reply:
Thanks